content/blog/2018-04-22-essurvey.md

```diff
@@ -0,0 +1,337 @@
+    library(tidyverse)
+
+Downloading the ESS data requires validating your email everytime you
+download data. We can set our email as an environment variable with
+`set_email`.
+
+    set_email("your@email.com")
+
+Once that's executed you can delete the previous line and any `import_*`
+call will look for the email automatically, stored as an environment
+variable.
+
+Exploring country data
+----------------------
+
+Let's suppose you don't know which countries or rounds (waves) are
+available for the ESS. Then the `show_*` family of functions is your
+friend.
+
+To find out which countries have participated you can use
+`show_countries()`
+
+    show_countries()
+
+    ##  [1] "Albania"            "Austria"            "Belgium"           
+    ##  [4] "Bulgaria"           "Croatia"            "Cyprus"            
+    ##  [7] "Czech Republic"     "Denmark"            "Estonia"           
+    ## [10] "Finland"            "France"             "Germany"           
+    ## [13] "Greece"             "Hungary"            "Iceland"           
+    ## [16] "Ireland"            "Israel"             "Italy"             
+    ## [19] "Kosovo"             "Latvia"             "Lithuania"         
+    ## [22] "Luxembourg"         "Netherlands"        "Norway"            
+    ## [25] "Poland"             "Portugal"           "Romania"           
+    ## [28] "Russian Federation" "Slovakia"           "Slovenia"          
+    ## [31] "Spain"              "Sweden"             "Switzerland"       
+    ## [34] "Turkey"             "Ukraine"            "United Kingdom"
+
+This function actually looks up the countries in the ESS website. If new
+countries enter, this will automatically grab those countries as well.
+Let's check out Spain. How many rounds has Spain participated in? We can
+use `show_country_rounds()`
+
+    sp_rnds <- show_country_rounds("Spain")
+    sp_rnds
+
+    ## [1] 1 2 3 4 5 6 7
+
+Note that country names are case sensitive. Use the exact name printed
+out by `show_countries()`.
+
+Using this information, we can download those specific rounds easily
+with `import_country`.
+
+Note: Since `essurvey 1.0.0` all `ess_*` functions have been deprecated
+in favour of the `import_*` and `download_*` functions.
+
+    spain <-
+      import_country(
+        country = "Spain",
+        rounds = 1:7
+        )
+
+`spain` will now be a list of `length(rounds)` containing a data frame
+for each round. The `import_*` family is concerned with downloading the
+data and thus always returns a list containing data frames unless only
+one round is specified, in which it returns a `tibble`. Conversely, the
+`show_*` family grabs information from the ESS website and always
+returns vectors.
+
+To download all rounds for a country automatically you can use
+`import_all_cntrounds`.
+
+    import_all_cntrounds("Spain")
+
+Getting back to our example, ESS related datasets have missing
+categories coded differently between questions. For example, questions

```

Comment:
"coded differently between questions." Relating to my request for definitions above, I didn't understand what this meant. Similar with "scales" here. Had to infer, but readers without survey experience might really have difficulty here